src/shared/urls.ts

```diff
@@ -5,7 +5,7 @@ const trimStart = (str: string): string => {
   return str.replace(/^\s+/, '');
 };
 
-const SUPPORTED_PROTOCOLS = ['http:', 'https:', 'ftp:', 'mailto:', 'about:'];
+const SUPPORTED_PROTOCOLS = ['http:', 'https:', 'ftp:', 'mailto:', 'about:', 'localhost:'];
```

Comment:
Please consider localhost without a port.  Do not add here because `localhost` is not a protocol.

Add condition (or private hleper method) to checks whether the url is a localhost:

```typescript
const isLocalhost = (url: string): boolean => {
  if (url === 'localhost') {
    return true;
  }
  let parts = url.split(':');
  return parts.length == 2 && !isNaN(parts[1]);
}

// ...

  if ((isLocalhost(keywords) || keywords.includes('.')) && !keywords.includes(' ')) {
    return 'http://' + keywords;
  }
```

Reply:
OK. I'll fix these cases. Furthermore, since input for :open follows in a way of "url keyword keyword", is it necessary to deal with url contains %s or {} ? For example, an input like "https://google.com/search?q={} keyword to search".

Reply:
A placeholder is used in only configuration of the search engines but not the console.

The `keyword` is a user's input from the console.  It never contains a placeholder `{}`.